test/dsl/dsl_options.jl

```diff
@@ -339,3 +339,236 @@ let
+    @test oprob_dsl[:X] == oprob_prog[:X]
+    @test oprob_dsl[:Y] == oprob_prog[:Y]
+    @test sol_dsl[:X] == sol_prog[:X]
+    @test sol_dsl[:Y] == sol_prog[:Y]
+end
+
+# Tests for complicated observable formula.
+# Tests using a single observable (without begin/end statement).
+# Tests using observable component not part of reaction.
+# Tests using parameters in observables formula.
+let 
+    rn = @reaction_network begin
+        @parameters op_1 op_2
+        @species X4(t)
+        @observables X ~ X1^2 + op_1*(X2 + 2X3) + X1*X4/op_2 + p        
+        (p,d), 0 <--> X1
+        (k1,k2), X1 <--> X2
+        (k3,k4), X2 <--> X3
+    end
+    
+    u0 = Dict([:X1 => 1.0, :X2 => 2.0, :X3 => 3.0, :X4 => 4.0])
+    ps = Dict([:p => 1.0, :d => 0.2, :k1 => 1.5, :k2 => 1.5, :k3 => 5.0, :k4 => 5.0, :op_1 => 1.5, :op_2 => 1.5])
+
+    oprob = ODEProblem(rn, u0, (0.0, 1000.0), ps)
+    sol = solve(oprob, Tsit5())
+
+    @test sol[:X][1] == u0[:X1]^2 + ps[:op_1]*(u0[:X2] + 2*u0[:X3]) + u0[:X1]*u0[:X4]/ps[:op_2] + ps[:p]  
+end
+
+# Checks that ivs are correctly found.
+let
+    rn = @reaction_network begin
+        @ivs t x y
+        @species V1(t) V2(t,x) V3(t, y) W1(t) W2(t, y)
+        @observables begin
+            V ~ V1 + 2V2 + 3V3
+            W ~ W1 + W2
+        end
+    end
+    V,W = getfield.(observed(rn), :lhs)
+    @test isequal(arguments(ModelingToolkit.unwrap(V)), Any[rn.iv, rn.sivs[1], rn.sivs[2]])
+    @test isequal(arguments(ModelingToolkit.unwrap(W)), Any[rn.iv, rn.sivs[2]])
+end
+
+# Checks that metadata is written properly.
+let
+    rn = @reaction_network rn_observed begin
+        @observables (X, [description="my_description"]) ~ X1 + X2
+        k, 0 --> X1 + X2
+    end
+    @test getdescription(observed(rn)[1].lhs) == "my_description"
+end
+
+# Declares observables implicitly/explicitly.
+# Cannot test `isequal(rn1, rn2)` because the two sets of observables have some obscure Symbolics
+# substructure that is different.
+let 
+    # Basic case.
+    rn1 = @reaction_network rn_observed begin
+        @observables X ~ X1 + X2
+        k, 0 --> X1 + X2
+    end
+    rn2 = @reaction_network rn_observed begin
+        @variables X(t)
+        @observables X ~ X1 + X2
+        k, 0 --> X1 + X2
+    end
+    @test isequal(observed(rn1)[1].rhs, observed(rn2)[1].rhs)
+    @test isequal(observed(rn1)[1].lhs.metadata, observed(rn2)[1].lhs.metadata)
+    @test isequal(states(rn1), states(rn2))
+
+    # Case with metadata.
+    rn3 = @reaction_network rn_observed begin
+        @observables (X,  [description="description"]) ~ X1 + X2
+        k, 0 --> X1 + X2
+    end
+    rn4 = @reaction_network rn_observed begin
+        @variables X(t) [description="description"]
+        @observables X ~ X1 + X2
+        k, 0 --> X1 + X2
+    end
+    @test isequal(observed(rn3)[1].rhs, observed(rn4)[1].rhs)
+    @test isequal(observed(rn3)[1].lhs.metadata, observed(rn4)[1].lhs.metadata)
+    @test isequal(states(rn3), states(rn4))
+end
```

Comment:
Do you have a version explicitly declaring an observable as a species (with metadata) too?

Reply:
added now